Fastify Typescript RouteSchema missing headers?: JSONSchema

Hello

Thank you for the great Fastify project. We're moving from Express to Fastify, because of async/await support and schema driven approach.

While using fastify v2 with typescript the following code:

```js
  const endpointsSchemas: fastify.RouteSchema = {
    querystring: '#schema-id-1#',
    headers: '#schema-id-2#',
    response: {
      200: '#schema-id-3#',
    },
  };
```

produces ts error:

```sh
error TS2322: Type '{ querystring: string; headers: string; response: { 200: string; }; }' is not assignable to type 'RouteSchema'.
  Object literal may only specify known properties, and 'headers' does not exist in type 'RouteSchema'.
         headers: '#schema-id-2#',
```

fix seems simple:
instead of https://github.com/fastify/fastify/blob/master/fastify.d.ts#L185
```js
  interface RouteSchema {
    body?: JSONSchema
    querystring?: JSONSchema
    params?: JSONSchema
    response?: {
      [code: number]: JSONSchema,
      [code: string]: JSONSchema
    }
  }
```

it should be

```js
  interface RouteSchema {
    body?: JSONSchema
    querystring?: JSONSchema
    params?: JSONSchema
    headers?: JSONSchema
    response?: {
      [code: number]: JSONSchema,
      [code: string]: JSONSchema
    }
  }
```

If that is correct I'd be happy to open PR with the fix.

Meanwhile we're using self declared type.

--- a/fastify.d.ts
+++ b/fastify.d.ts
@@ -170,6 +170,7 @@ declare namespace fastify {
     body?: JSONSchema
     querystring?: JSONSchema
     params?: JSONSchema
+    headers?: JSONSchema
     response?: {
       [code: number]: JSONSchema,
       [code: string]: JSONSchema

--- a/test/types/index.ts
+++ b/test/types/index.ts
@@ -138,27 +138,41 @@ server.addHook('onClose', (instance, done) => {
 server.addHook('onRoute', (opts) => {
 })
 
-const opts: fastify.RouteShorthandOptions<http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse> = {
-  schema: {
-    response: {
-      200: {
-        type: 'object',
-        properties: {
-          hello: {
-            type: 'string'
-          }
+const schema: fastify.RouteSchema = {
+  body: {
+    type: 'object'
+  },
+  querystring: {
+    type: 'object'
+  },
+  params: {
+    type: 'object'
+  },
+  headers: {
+    type: 'object'
+  },
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        hello: {
+          type: 'string'
         }
-      },
-      '2xx': {
-        type: 'object',
-        properties: {
-          hello: {
-            type: 'string'
-          }
+      }
+    },
+    '2xx': {
+      type: 'object',
+      properties: {
+        hello: {
+          type: 'string'
         }
       }
     }
-  },
+  }
+}
+
+const opts: fastify.RouteShorthandOptions<http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse> = {
+  schema,
   preValidation: [
     (request, reply, next) => {
       request.log.info(`pre validation for "${request.raw.url}" ${request.id}`)

